featurebyte/storage/base.py

```diff
@@ -67,22 +72,58 @@ async def try_delete_if_exists(self, remote_path: Path) -> None:
         except FileNotFoundError:
             pass
 
-    @abstractmethod
-    async def get(self, remote_path: Path, local_path: Path) -> None:
+    async def get(self, remote_path: Path, local_path: Path, cache_key: Any = None) -> None:
         """
-        Download file from storage to local path
+        Download file from storage to local path with caching if cache_key is provided
 
         Parameters
         ----------
         remote_path: Path
             Path of remote file to be downloaded
         local_path: Path
             Path to stored downloaded file
+        cache_key: Any
+            Cache key for storing downloaded file (if provided, the result will be cached).
+            Please do not use cache_key for large files.
 
-        Raises
-        ------
-        FileNotFoundError
-            Remote file does not exist
+        Returns
+        -------
+        None
+        """
+        if cache_key is None:
+            return await self._get(remote_path, local_path)
+
+        # cache hit
+        try:
+            value = self._cache[cache_key]
+            async with aiofiles.open(local_path, "wb") as file_obj:
+                await file_obj.write(value)
+            return None
+        except KeyError:
+            pass
+
+        # cache miss
+        await self._get(remote_path, local_path)
+        try:
+            async with aiofiles.open(local_path, "rb") as file_obj:
+                value = await file_obj.read()
+
+            async with self._cache_lock:
+                self._cache.setdefault(cache_key, value)
+        except ValueError:
+            pass  # value too large
```

Comment:
Is this a limit that can be configured?

Reply:
This is based on the implementation in `cachedtools` library:
https://github.com/tkem/cachetools/blob/master/src/cachetools/__init__.py#L781-L844

Reply:
This is controlled by the cache size (https://github.com/tkem/cachetools/blob/master/src/cachetools/__init__.py#L72-L86).